pkg/vere/io/http.c

```diff
@@ -302,6 +304,71 @@ _http_heds_from_noun(u3_noun hed)
   return hed_u;
 }
 
+/* _http_req_is_auth(): returns c3y if rec_u contains a valid auth cookie
+*/
+static c3_o
+_http_req_is_auth(u3_hfig* fig_u, h2o_req_t* rec_u)
+{
+  //  try to find a cookie header
+  //
+  h2o_iovec_t coo_u = {NULL, 0};
+  {
+    //TODO  http2 allows the client to put multiple 'cookie' headers,
+    //      runtime should support that once eyre does too.
+    ssize_t hin_i = h2o_find_header_by_str(&rec_u->headers, "cookie", 6, -1);
+    if ( hin_i != -1 ) {
+      coo_u = rec_u->headers.entries[hin_i].value;
+    }
+  }
+
+  //  if there is no cookie header, it can't possibly be authenticated
+  //
+  if ( NULL == coo_u.base ) {
+    return c3n;
+  }
+  //  if there is a cookie, see if it contains a valid auth token
+  //
+  else {
+    c3_c* key_c = fig_u->key_c;
+    c3_c  val_c[128];
+    c3_y  val_y = 0;
+    c3_y    i_y = 0;
+    c3_y    j_y = 0;
```

Comment:
Is the max length of `coo_u.base` less than or equal to `255`? The reason I ask is because it's unclear to me whether this code avoids overflowing its index variables i.e. `i_y`, `j_y`. On a similar note, is the max auth token length `128`?

Reply:
Yes, auth tokens are 128 bits of entropy, encoded in `@uv` representation. In practice this'll be 33 characters or thereabouts, so 128 characters is plenty.

Good catch on the overflows though. I bumped them up to `c3_w` which is much more reasonable here. What's the appropriate way to say "if `coo_u.len` is larger than `c3_w`'s max value, consider it unauthed"?

Reply:
> Yes, auth tokens are 128 bits of entropy, encoded in @uv representation. In practice this'll be 33 characters or thereabouts, so 128 characters is plenty.

Great. Thanks for clarifying.

> What's the appropriate way to say "if coo_u.len is larger than c3_w's max value, consider it unauthed"?

Do you think it makes sense to bump the types up to `size_t`? That way, they can't possibly overflow. Another approach would be to check `coo_u.len` against the hard-coded max of `c3_w` and include a comment indicating the hard-coded max must be updated if the types of `i_w` and `j_w` change.